src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -995,17 +995,18 @@ async Task ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationTok
             : IPAddressesToEndpoints(await TaskTimeoutAndCancellation.ExecuteAsync(GetHostAddressesAsync, timeout, cancellationToken).ConfigureAwait(false),
                 Port);
 
-        // Give each IP an equal share of the remaining time
```

Comment:
What if we simplify things a bit, like this:

```csharp
// Give each IP an equal share of the remaining time
var perIpTimespan = default(TimeSpan);
if (timeout.IsSet)
	perIpTimespan = new TimeSpan(timeout.CheckAndGetTimeLeft().Ticks / endpoints.Length);

for (var i = 0; i < endpoints.Length; i++)
{
	var perIpTimeout = new NpgsqlTimeout(perIpTimespan);
	Debug.Assert(timeout.IsSet == perIpTimeout.IsSet);
```

Reply:
Looks good to me, at least each endpoint gets exactly the same amount of time and I think is closer to how the blocking version works.